## Data scraping

In [6]:
import pandas as pd
import requests
from bs4 import BeautifulSoup

In [7]:
scores = requests.get("https://www.walkscore.com/cities-and-neighborhoods/").text
soup = BeautifulSoup(scores)

In [8]:
#scraping the headers

headers = soup.select("th")
for header in headers:
    print(header.text)

City
State
Walk Score
Transit Score
Bike Score
Population


In [9]:
#scraping cities and saving them into a list

city_info = []

cities = soup.select(".city a")
for city in cities:
    city_info.append(city.text)
    
city_info

['Gatineau',
 'Longueuil',
 'Québec',
 'Montréal',
 'Laval',
 'Windsor',
 'Kitchener',
 'Ottawa',
 'Vaughan',
 'Hamilton',
 'Mississauga',
 'Brampton',
 'Toronto',
 'London',
 'Markham',
 'Winnipeg',
 'Saskatoon',
 'Calgary',
 'Edmonton',
 'Surrey',
 'Burnaby',
 'Vancouver',
 'Los Angeles',
 'San Diego',
 'San Jose',
 'San Francisco',
 'Fresno',
 'Sacramento',
 'Long Beach',
 'Oakland',
 'Bakersfield',
 'Anaheim',
 'Santa Ana',
 'Riverside',
 'Stockton',
 'Chula Vista',
 'Fremont',
 'Irvine',
 'San Bernardino',
 'Detroit',
 'Chicago',
 'Seattle',
 'Spokane',
 'Portland',
 'New York',
 'Buffalo',
 'Rochester',
 'Phoenix',
 'Tucson',
 'Mesa',
 'Chandler',
 'Glendale',
 'Scottsdale',
 'Gilbert',
 'New Orleans',
 'Baton Rouge',
 'Jacksonville',
 'Miami',
 'Tampa',
 'St. Petersburg',
 'Orlando',
 'Hialeah',
 'Kansas City',
 'St. Louis',
 'Atlanta',
 'Virginia Beach',
 'Norfolk',
 'Chesapeake',
 'Arlington',
 'Richmond',
 'Baltimore',
 'Boston',
 'Columbus',
 'Cleveland',
 'Cincinnati',
 '

In [10]:
#scraping states and saving them into a list

state_info = []

states = soup.select("tbody .state")
for state in states:
    state_info.append(state.text)
    
state_info

['CA-QC',
 'CA-QC',
 'CA-QC',
 'CA-QC',
 'CA-QC',
 'CA-ON',
 'CA-ON',
 'CA-ON',
 'CA-ON',
 'CA-ON',
 'CA-ON',
 'CA-ON',
 'CA-ON',
 'CA-ON',
 'CA-ON',
 'CA-MB',
 'CA-SK',
 'CA-AB',
 'CA-AB',
 'CA-BC',
 'CA-BC',
 'CA-BC',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'CA',
 'MI',
 'IL',
 'WA',
 'WA',
 'OR',
 'NY',
 'NY',
 'NY',
 'AZ',
 'AZ',
 'AZ',
 'AZ',
 'AZ',
 'AZ',
 'AZ',
 'LA',
 'LA',
 'FL',
 'FL',
 'FL',
 'FL',
 'FL',
 'FL',
 'MO',
 'MO',
 'GA',
 'VA',
 'VA',
 'VA',
 'VA',
 'VA',
 'MD',
 'MA',
 'OH',
 'OH',
 'OH',
 'OH',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'TX',
 'OK',
 'OK',
 'AL',
 'AL',
 'CO',
 'CO',
 'CO',
 'IN',
 'IN',
 'NC',
 'NC',
 'NC',
 'NC',
 'NC',
 'NC',
 'WI',
 'WI',
 'NJ',
 'NJ',
 'MN',
 'MN',
 'TN',
 'TN',
 'IA',
 'KY',
 'KY',
 'HI',
 'NV',
 'NV',
 'NV',
 'NV',
 'NV',
 'PA',
 'PA',
 'NE',
 'NE',
 'KS',
 'AK',
 'ID',
 'NM',
 'DC']

In [15]:
#scraping the walkscore of individual cities and saving them into a list

walkscore_info = []
walkscores = soup.select("tbody .score")
for scores in walkscores:
    walkscore_info.append(scores.text)

walkscore_info

['37.4',
 '54.4',
 '44.6',
 '65.4',
 '43.1',
 '42.4',
 '44.6',
 '45.1',
 '34.8',
 '49.6',
 '48.9',
 '42.5',
 '61.0',
 '38.6',
 '34.5',
 '48.5',
 '45.7',
 '39.1',
 '40.2',
 '46.1',
 '60.1',
 '79.8',
 '68.6',
 '53.3',
 '50.5',
 '88.7',
 '46.6',
 '49.0',
 '73.3',
 '75.3',
 '37.3',
 '55.7',
 '67.1',
 '42.7',
 '43.7',
 '45.6',
 '49.9',
 '43.2',
 '45.0',
 '51.1',
 '77.2',
 '74.4',
 '49.2',
 '67.3',
 '88.0',
 '66.6',
 '60.7',
 '41.4',
 '43.2',
 '37.9',
 '35.4',
 '40.2',
 '31.9',
 '29.0',
 '58.0',
 '39.1',
 '25.6',
 '76.6',
 '49.5',
 '43.0',
 '41.1',
 '67.9',
 '35.3',
 '65.7',
 '47.7',
 '33.1',
 '45.8',
 '21.3',
 '71.2',
 '50.9',
 '64.3',
 '82.8',
 '41.2',
 '57.1',
 '49.1',
 '46.4',
 '47.5',
 '36.9',
 '46.0',
 '41.7',
 '34.9',
 '40.4',
 '38.1',
 '40.3',
 '40.5',
 '36.8',
 '39.0',
 '40.2',
 '44.8',
 '34.1',
 '39.0',
 '33.2',
 '25.8',
 '61.2',
 '36.1',
 '42.5',
 '31.0',
 '31.7',
 '26.4',
 '31.3',
 '29.4',
 '22.5',
 '30.3',
 '21.4',
 '61.5',
 '49.7',
 '75.9',
 '86.6',
 '71.4',
 '60.4',
 '35.0',
 

In [12]:
#scraping the transit score of individual cities and saving them into a list

transitscore_info = []
transitscores = soup.select("tbody .tsc")
for transitscore in transitscores:
    transitscore_info.append(transitscore.text)

transitscore_info

['40.8',
 '52.5',
 '46.9',
 '67.0',
 '46.3',
 '37.9',
 '46.8',
 '49.5',
 '42.8',
 '45.4',
 '56.0',
 '53.3',
 '78.2',
 '45.0',
 '49.3',
 '51.2',
 '45.0',
 '49.5',
 '48.6',
 '47.1',
 '66.8',
 '74.4',
 '52.9',
 '37.3',
 '40.1',
 '77.1',
 '33.1',
 '33.9',
 '48.9',
 '56.6',
 '24.6',
 '33.9',
 '43.3',
 '29.9',
 '25.4',
 '35.4',
 '36.3',
 '25.7',
 '30.9',
 '35.7',
 '65.0',
 '59.6',
 '36.0',
 '49.3',
 '88.6',
 '47.5',
 '41.3',
 '35.6',
 '34.5',
 '27.0',
 '24.4',
 '33.5',
 '23.3',
 '15.1',
 '43.9',
 '--',
 '20.8',
 '57.0',
 '31.3',
 '33.5',
 '32.7',
 '39.0',
 '25.5',
 '42.6',
 '43.7',
 '20.7',
 '35.7',
 '13.9',
 '58.6',
 '--',
 '53.2',
 '72.4',
 '30.2',
 '43.9',
 '44.1',
 '--',
 '36.2',
 '30.9',
 '38.9',
 '34.8',
 '21.6',
 '28.0',
 '0.3',
 '26.6',
 '14.8',
 '--',
 '20.6',
 '29.2',
 '28.7',
 '16.9',
 '22.7',
 '21.5',
 '16.4',
 '44.6',
 '18.8',
 '35.2',
 '24.9',
 '21.7',
 '27.4',
 '29.4',
 '--',
 '--',
 '27.5',
 '--',
 '48.7',
 '34.9',
 '65.0',
 '70.5',
 '55.0',
 '46.4',
 '22.2',
 '21.7',
 '30.1'

In [14]:
#scraping the bike score of individual cities and saving them into a list

bikescore_info = []
bikescores = soup.select("tbody .bsc")
for bikescore in bikescores:
    bikescore_info.append(bikescore.text)

bikescore_info

['57.6',
 '69.6',
 '59.3',
 '72.6',
 '57.3',
 '52.3',
 '54.9',
 '63.6',
 '44.3',
 '50.3',
 '54.1',
 '54.8',
 '61.0',
 '52.8',
 '46.8',
 '61.1',
 '31.7',
 '50.1',
 '39.5',
 '55.9',
 '59.5',
 '78.9',
 '58.7',
 '43.0',
 '61.5',
 '72.3',
 '58.1',
 '66.9',
 '70.1',
 '65.5',
 '43.7',
 '52.3',
 '62.2',
 '48.9',
 '52.4',
 '41.2',
 '54.4',
 '69.5',
 '43.6',
 '52.1',
 '72.2',
 '71.1',
 '52.1',
 '82.7',
 '69.3',
 '62.8',
 '59.9',
 '55.9',
 '66.3',
 '59.6',
 '55.4',
 '56.5',
 '56.4',
 '52.6',
 '65.7',
 '44.3',
 '40.5',
 '64.0',
 '55.2',
 '58.5',
 '56.8',
 '50.6',
 '35.0',
 '57.7',
 '41.7',
 '41.7',
 '46.9',
 '34.7',
 '72.3',
 '50.8',
 '53.4',
 '69.4',
 '47.7',
 '52.3',
 '35.6',
 '45.6',
 '48.6',
 '44.5',
 '49.3',
 '54.0',
 '38.6',
 '42.0',
 '39.5',
 '42.1',
 '52.0',
 '39.6',
 '42.5',
 '40.6',
 '42.0',
 '40.2',
 '47.2',
 '30.7',
 '32.9',
 '71.8',
 '44.7',
 '53.3',
 '42.8',
 '41.0',
 '31.3',
 '39.3',
 '32.2',
 '29.2',
 '37.6',
 '29.7',
 '57.6',
 '65.5',
 '51.1',
 '63.9',
 '83.5',
 '64.0',
 '41.2',
 

In [16]:
#scraping the population of individual cities and saving them into a list

pop_info = []
population = soup.select("tbody .pop")
for pop in population:
    pop_info.append(pop.text)

pop_info

['265,349',
 '231,409',
 '516,622',
 '1,649,519',
 '401,553',
 '210,891',
 '219,153',
 '883,391',
 '288,301',
 '519,949',
 '713,443',
 '523,911',
 '2,615,060',
 '366,151',
 '301,709',
 '663,617',
 '222,189',
 '1,096,833',
 '812,201',
 '468,251',
 '223,218',
 '603,502',
 '3,792,621',
 '1,307,402',
 '945,942',
 '805,235',
 '494,665',
 '466,488',
 '462,257',
 '390,724',
 '347,483',
 '336,265',
 '324,528',
 '303,871',
 '291,707',
 '243,916',
 '214,089',
 '212,375',
 '209,924',
 '713,777',
 '2,695,598',
 '608,660',
 '208,916',
 '583,776',
 '8,175,133',
 '261,310',
 '210,565',
 '1,445,632',
 '520,116',
 '439,041',
 '236,123',
 '226,721',
 '217,385',
 '208,453',
 '343,829',
 '229,493',
 '821,784',
 '399,457',
 '335,709',
 '244,769',
 '238,300',
 '224,669',
 '459,787',
 '319,294',
 '420,003',
 '437,994',
 '242,803',
 '222,209',
 '207,627',
 '204,214',
 '620,961',
 '617,594',
 '787,033',
 '396,815',
 '296,943',
 '287,208',
 '2,099,451',
 '1,327,407',
 '1,197,816',
 '790,390',
 '741,206',
 '649,

In [17]:
#pulling the different lists above into one common dictionary in order to turn it into a dataframe

info = {'city': city_info,
      'state': state_info,
       'walk_score': walkscore_info,
       'transit_score': transitscore_info,
       'bike_score': bikescore_info,
       'population': pop_info}

I took the manual route to build this dictionary because I couldn't figure out how to do clean the data when I got the result from one single for loop. For instance, one version would get me all the attributes of one city as a single string. While I could use split() to add comma between the different attributes, it didn't work for city with a space in its name like "New York." It would become "New," "York" — which doesn't help with building the dataframe. Another verison would give me attributes as individuals, and I couldn't seem to figure out how to group them by their relevant city.

## Data analysis: Canada

In [18]:
#converting the dictionary above into a dataframe for analysis and saving it as a new csv

table = pd.DataFrame(info, columns = ['city', 'state', 'walk_score', 'transit_score', 'bike_score', 'population'])
table.to_csv("walkable_cities.csv")

In [19]:
import pandas as pd

df = pd.read_csv('walkable_cities.csv')
canadian_cities = df[df['state'].str.contains('CA-')]
canadian_cities.to_csv("canadian_cities.csv")

In [20]:
canada = pd.read_csv('canadian_cities.csv')
len(canada)

#22 canadian cities

22

I'm pulling the Canada-only data into a separate csv for easier analysis. 

In [42]:
CA_walkscore = canada.walk_score.mean().round(2)
print(CA_walkscore)
CA_transitscore = canada.transit_score.mean().round(2)
print(CA_transitscore)

47.56
51.83


In [41]:
print(canada.query('walk_score < 47.56').count()) #14
print(canada.query('transit_score < 51.83').count()) #15

Unnamed: 0.1     14
Unnamed: 0       14
city             14
state            14
walk_score       14
transit_score    14
bike_score       14
population       14
dtype: int64
Unnamed: 0.1     15
Unnamed: 0       15
city             15
state            15
walk_score       15
transit_score    15
bike_score       15
population       15
dtype: int64


In [23]:
#The 5 Canadian cities with the best walk score

canada.sort_values('walk_score', ascending=False).head()

Unnamed: 0.1  Unnamed: 0       city  state  walk_score  transit_score  \
21            21          21  Vancouver  CA-BC        79.8           74.4   
3              3           3  Montréal  CA-QC        65.4           67.0   
12            12          12    Toronto  CA-ON        61.0           78.2   
20            20          20    Burnaby  CA-BC        60.1           66.8   
1              1           1  Longueuil  CA-QC        54.4           52.5   

    bike_score population  
21        78.9    603,502  
3         72.6  1,649,519  
12        61.0  2,615,060  
20        59.5    223,218  
1         69.6    231,409

In [24]:
#The 5 Canadian cities with the worst walk score

canada.sort_values('walk_score', ascending=False).tail()

Unnamed: 0.1  Unnamed: 0      city  state  walk_score  transit_score  \
17            17          17   Calgary  CA-AB        39.1           49.5   
13            13          13    London  CA-ON        38.6           45.0   
0              0           0  Gatineau  CA-QC        37.4           40.8   
8              8           8   Vaughan  CA-ON        34.8           42.8   
14            14          14   Markham  CA-ON        34.5           49.3   

    bike_score population  
17        50.1  1,096,833  
13        52.8    366,151  
0         57.6    265,349  
8         44.3    288,301  
14        46.8    301,709

In [25]:
#The 5 Canadian cities with the best transit score

canada.sort_values('transit_score', ascending=False).head()

Unnamed: 0.1  Unnamed: 0         city  state  walk_score  transit_score  \
12            12          12      Toronto  CA-ON        61.0           78.2   
21            21          21    Vancouver  CA-BC        79.8           74.4   
3              3           3    Montréal  CA-QC        65.4           67.0   
20            20          20      Burnaby  CA-BC        60.1           66.8   
10            10          10  Mississauga  CA-ON        48.9           56.0   

    bike_score population  
12        61.0  2,615,060  
21        78.9    603,502  
3         72.6  1,649,519  
20        59.5    223,218  
10        54.1    713,443

In [45]:
#The 5 Canadian cities with the worst transit score

canada.sort_values('transit_score', ascending=False).tail()

Unnamed: 0.1  Unnamed: 0       city  state  walk_score  transit_score  \
13            13          13     London  CA-ON        38.6           45.0   
16            16          16  Saskatoon  CA-SK        45.7           45.0   
8              8           8    Vaughan  CA-ON        34.8           42.8   
0              0           0   Gatineau  CA-QC        37.4           40.8   
5              5           5    Windsor  CA-ON        42.4           37.9   

    bike_score  population  
13        52.8      366151  
16        31.7      222189  
8         44.3      288301  
0         57.6      265349  
5         52.3      210891

## Data analysis: US 

In [29]:
#saving a separate csv for just American cities

us_only = table.drop(table.index[0:22])
us_only.to_csv('us_only.csv')

In [30]:
us = pd.read_csv('us_only.csv')
len(us)

108

I removed the Canada-only data to get the US-only data and saving it as a separate CSV for easier analysis and visualization.

In [31]:
#The 10 American cities with the best walkscore

us.sort_values('walk_score', ascending=False).head(10)

Unnamed: 0             city state  walk_score transit_score  bike_score  \
3            25    San Francisco    CA        88.7          77.1        72.3   
22           44         New York    NY        88.0          88.6        69.3   
85          107      Jersey City    NJ        86.6          70.5        63.9   
49           71           Boston    MA        82.8          72.4        69.4   
18           40          Chicago    IL        77.2          65.0        72.2   
107         129  Washington D.C.    DC        76.7          68.7        69.5   
35           57            Miami    FL        76.6          57.0        64.0   
84          106           Newark    NJ        75.9          65.0        51.1   
7            29          Oakland    CA        75.3          56.6        65.5   
99          121     Philadelphia    PA        74.8          67.0        66.7   

    population  
3      805,235  
22   8,175,133  
85     247,597  
49     617,594  
18   2,695,598  
107    601,723  
35     399,457  
84     277,140  
7      390,724  
99   1,526,006

In [32]:
#The 10 American cities with the worst walkscore

us.sort_values('walk_score', ascending=False).tail(10)

Unnamed: 0                city state  walk_score transit_score  \
95         117           Henderson    NV        30.0          24.6   
78         100          Greensboro    NC        29.4            --   
31          53             Gilbert    AZ        29.0          15.1   
89         111  Nashville-Davidson    TN        28.8          21.7   
76          98           Charlotte    NC        26.4          27.4   
70          92          Montgomery    AL        25.8          16.4   
34          56        Jacksonville    FL        25.6          20.8   
79         101       Winston-Salem    NC        22.5            --   
81         103        Fayetteville    NC        21.4            --   
45          67          Chesapeake    VA        21.3          13.9   

    bike_score population  
95        43.5    257,729  
78        32.2    269,666  
31        52.6    208,453  
89        29.7    601,222  
76        31.3    731,424  
70        32.9    205,764  
34        40.5    821,784  
79        29.2    229,617  
81        29.7    200,564  
45        34.7    222,209

In [33]:
print(us.walk_score.mean().round(2)) 

47.8


Unsurprisingly, the cities with the best walk score are mostly in the Northeast while those at the bottom are in the South and Southwest. I also calculated the average walkscore (47.8) for the 108 US cities, which is actually just a hair higher than the average walkscore (47.6) for the 22 Canadian cities. 

In [34]:
bottom_score = us.query('walk_score < 47.8')
bottom_score.to_csv('bottom_walkscore.csv')
len(bottom_score)

#There are 67 cities with below-average walk score, which is more than 60% of the cities included in this ranking. 

66

In [35]:
bottom_walkscore = pd.read_csv('bottom_walkscore.csv')
bottom_walkscore.sort_values('walk_score', ascending=True)

Unnamed: 0.1  Unnamed: 0           city state  walk_score transit_score  \
22            45          67     Chesapeake    VA        21.3          13.9   
51            81         103   Fayetteville    NC        21.4            --   
49            79         101  Winston-Salem    NC        22.5            --   
15            34          56   Jacksonville    FL        25.6          20.8   
41            70          92     Montgomery    AL        25.8          16.4   
..           ...         ...            ...   ...         ...           ...   
27            56          78         Dallas    TX        46.0          38.9   
24            53          75         Toledo    OH        46.4            --   
0              4          26         Fresno    CA        46.6          33.1   
25            54          76        Houston    TX        47.5          36.2   
19            42          64        Atlanta    GA        47.7          43.7   

    bike_score population  
22        34.7    222,209  
51        29.7    200,564  
49        29.2    229,617  
15        40.5    821,784  
41        32.9    205,764  
..         ...        ...  
27        49.3  1,197,816  
24        45.6    287,208  
0         58.1    494,665  
25        48.6  2,099,451  
19        41.7    420,003  

[66 rows x 8 columns]

In [36]:
us.query('walk_score < 47.8').groupby('state').count()

Unnamed: 0  city  walk_score  transit_score  bike_score  population
state                                                                     
AK              1     1           1              1           1           1
AL              2     2           2              2           2           2
AZ              7     7           7              7           7           7
CA              7     7           7              7           7           7
CO              2     2           2              2           2           2
FL              3     3           3              3           3           3
GA              1     1           1              1           1           1
IA              1     1           1              1           1           1
ID              1     1           1              1           1           1
IN              2     2           2              2           2           2
KS              1     1           1              1           1           1
KY              2     2           2              2           2           2
LA              1     1           1              1           1           1
MO              1     1           1              1           1           1
NC              6     6           6              6           6           6
NE              1     1           1              1           1           1
NM              1     1           1              1           1           1
NV              4     4           4              4           4           4
OH              2     2           2              2           2           2
OK              2     2           2              2           2           2
TN              2     2           2              2           2           2
TX             13    13          13             13          13          13
VA              3     3           3              3           3           3

I calculated how many out of the 108 US cities have below-average walkscores, and there are 66 of them. Notably, Texas leads this group with 13 cities, which are all of its cities included in this ranking. 

In [37]:
us.query('transit_score == "--"')
updated_us = us.drop(us.index[[33,47,53,63,78,79,81]])
updated_us['transit_score'] = updated_us['transit_score'].astype(float)
updated_us.transit_score.mean().round(2)

35.98

In [43]:
print(updated_us.query('transit_score < 35.98').count())

#there are 63 US cities with below-average transit score

Unnamed: 0       63
city             63
state            63
walk_score       63
transit_score    63
bike_score       63
population       63
dtype: int64


I removed the rows with "--" as their transit score, so I could actually calculate the average transit score for the 101 US cities featured and rank the cities. I also had to convert the data into float in order to do the calculation.

In [38]:
#The 10 worst US cities for transit score

updated_us.sort_values('transit_score', ascending=True).head(10)

Unnamed: 0              city state  walk_score  transit_score  \
60           82         Arlington    TX        38.1            0.3   
45           67        Chesapeake    VA        21.3           13.9   
62           84             Plano    TX        40.5           14.8   
31           53           Gilbert    AZ        29.0           15.1   
70           92        Montgomery    AL        25.8           16.4   
67           89     Oklahoma City    OK        34.1           16.9   
72           94  Colorado Springs    CO        36.1           18.8   
103         125           Wichita    KS        34.8           19.7   
64           86           Lubbock    TX        39.0           20.6   
43           65    Virginia Beach    VA        33.1           20.7   

     bike_score population  
60         39.5    365,438  
45         34.7    222,209  
62         52.0    259,841  
31         52.6    208,453  
70         32.9    205,764  
67         40.2    579,999  
72         44.7    416,427  
103        43.9    382,368  
64         42.5    229,573  
43         41.7    437,994

In [39]:
#The 10 best US cities for transit score

updated_us.sort_values('transit_score', ascending=False).head(10)

Unnamed: 0             city state  walk_score  transit_score  bike_score  \
22           44         New York    NY        88.0           88.6        69.3   
3            25    San Francisco    CA        88.7           77.1        72.3   
49           71           Boston    MA        82.8           72.4        69.4   
85          107      Jersey City    NJ        86.6           70.5        63.9   
107         129  Washington D.C.    DC        76.7           68.7        69.5   
99          121     Philadelphia    PA        74.8           67.0        66.7   
18           40          Chicago    IL        77.2           65.0        72.2   
84          106           Newark    NJ        75.9           65.0        51.1   
19           41          Seattle    WA        74.4           59.6        71.1   
46           68        Arlington    VA        71.2           58.6        72.3   

    population  
22   8,175,133  
3      805,235  
49     617,594  
85     247,597  
107    601,723  
99   1,526,006  
18   2,695,598  
84     277,140  
19     608,660  
46     207,627

Unsurprisingly, the Northeast once again dominates the list for the 10 cities with the best transit scores. Meanwhile, cities in the South and Southwest are the majority for the bottom 10, with Texas having 3 cities alone. Notably, Arlington, TX has 0.3 for transit score aka no public transit at all. This got me creating a separate Texas csv for data vis below.

The average walk score (36) for the featured US cities is also far behind that of Canadian cities (51.8). 

In [40]:
texas = us.query('state == "TX"')
texas.to_csv('texas.csv')
texas

Unnamed: 0            city state  walk_score transit_score  bike_score  \
54          76         Houston    TX        47.5          36.2        48.6   
55          77     San Antonio    TX        36.9          30.9        44.5   
56          78          Dallas    TX        46.0          38.9        49.3   
57          79          Austin    TX        41.7          34.8        54.0   
58          80      Fort Worth    TX        34.9          21.6        38.6   
59          81         El Paso    TX        40.4          28.0        42.0   
60          82       Arlington    TX        38.1           0.3        39.5   
61          83  Corpus Christi    TX        40.3          26.6        42.1   
62          84           Plano    TX        40.5          14.8        52.0   
63          85          Laredo    TX        36.8            --        39.6   
64          86         Lubbock    TX        39.0          20.6        42.5   
65          87         Garland    TX        40.2          29.2        40.6   
66          88          Irving    TX        44.8          28.7        42.0   

   population  
54  2,099,451  
55  1,327,407  
56  1,197,816  
57    790,390  
58    741,206  
59    649,121  
60    365,438  
61    305,215  
62    259,841  
63    236,091  
64    229,573  
65    226,876  
66    216,290